pkg/controller/perconaservermongodb/balancer.go

```diff
@@ -103,10 +103,17 @@ func (r *ReconcilePerconaServerMongoDB) enableBalancerIfNeeded(ctx context.Conte
 	return nil
 }
 
+func (r *ReconcilePerconaServerMongoDB) disableBalancerIfNeeded(ctx context.Context, cr *api.PerconaServerMongoDB) error {
+	if s := cr.Spec.Sharding; !s.Enabled || s.Mongos.Size == 0 || cr.Spec.Unmanaged || s.Balancer.Enabled == nil || *s.Balancer.Enabled {
```

Comment:
we shouldn't disable balancer if cluster is unmanaged. if cluster is unmanaged we simply shouldn't touch it.

Reply:
This if condition is positive when `cr.Spec.Unmanaged == true`